src/ase/cw/model/OrderQueue.java

```diff
@@ -11,19 +11,44 @@
     private List<Order> loadedOrders;
     private OrderProducerListener listener;
     private int maxDelayTime = 1000;
+    private Thread orderQueueThread;
+    private boolean shouldPause = false;
+    private String name = "Order Queue";
+
 
     public OrderQueue(Vector<Order> loadedOrders, OrderProducerListener listener) {
         this.loadedOrders = loadedOrders;
         this.listener = listener;
+            if (orderQueueThread == null) {
```

Comment:
just sayin: this if is not needed, orderQueueThread is always null when you call the constructor